xgi/communities/spectral.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+from scipy.sparse.linalg import eigsh
+
+from ..core import Hypergraph
+from ..linalg.laplacian_matrix import normalized_hypergraph_laplacian
+
+from ..exception import XGIError
+
+__all__ = [
+    "spectral_clustering",
+]
+
+MAX_ITERATIONS = 10_000
+
+
+def spectral_clustering(H, k=None):
+    """Cluster into k-many groups using spectral techniques.
+
+    Compute a spectral clustering according to the heuristic suggested in [1].
+
+    Parameters
+    ----------
+    H : Hypergraph
+        Hypergraph
+    k : int, optional
+        Number of clusters to find. If unspecified, computes spectral gap.
+
+    Returns
+    -------
+    dict
+        A dictionary mapping node ids to their clusters. Clusters begin at 0.
+
+    Raises
+    ------
+    XGIError
+        If more groups are specified than nodes in the hypergraph.
+
+
+    References
+    ----------
+    .. [1] Zhou, D., Huang, J., & Schölkopf, B. (2006).
+        Learning with Hypergraphs: Clustering, Classification, and Embedding
+        Advances in Neural Information Processing Systems.
+
+    """
+    if k is None:
+        raise NotImplementedError(
+            "Choosing a number of clusters organically is currently unsupported. Please specify an integer value for paramater 'k'!"
+        )
```

Comment:
I'd say either we implement this `k=None` default option, or we remove the default value and these lines (until it's maybe implemented).